### ISSUE TITLE
Add support for registering custom `HttpMetricsTagProvider` in Stargate V2/REST

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/stargate#1565]
(note: follow-up to #1551)

Due to StargateV2 refactoring the mechanism used for adding HTTP metrics tag providers will not work (it is based on OSGi).
We need to re-connect this, but it seems that this should be part of work when moving to the new HTTP/Service framework (see discussion #1526).

This issue needs to be tackled before actual production deployments as tags are important for analytics, dashboards.

**Comment by tatu-at-datastax** [comment]
Will work on this as it will be needed for SGv2 Beta launch, for which web framework used by REST API will remain DropWizard.

**Comment by tatu-at-datastax** [comment]
Hmmh. Looks like we are not only missing tenant id but also most other tags.
I have tried to look for any differences in initialization code but cannot really find meaningful differences.
For some reason, while `/metrics` endpoint for SGv1 shows entries like:

```
# HELP http_server_requests_counter_total  
# TYPE http_server_requests_counter_total counter
http_server_requests_counter_total{error="false",module="restapi",} 1200000.0
http_server_requests_counter_total{error="false",module="authapi",} 1.0
```

nothing like that is exposed by SGv2 REST API. Not just no tags but no "http_server_*" metrics at all.

Yet there are default Jersey/DropWizard entries like:

```
# HELP sgv2_restapi_io_stargate_sgv2_restsvc_resources_Sgv2RowsResourceApi_createRow_total Generated from Dropwizard metric import (metric=sgv2-restapi.io.stargate.sgv2.restsvc.resources.Sgv2RowsResourceApi.createRow.total, type=com.codahale.metrics.Timer)
# TYPE sgv2_restapi_io_stargate_sgv2_restsvc_resources_Sgv2RowsResourceApi_createRow_total summary
sgv2_restapi_io_stargate_sgv2_restsvc_resources_Sgv2RowsResourceApi_createRow_total{quantile="0.5",} 0.0
...
```

**Comment by tatu-at-datastax** [comment]
Ok never mind. I made a mistake of looking at metrics before sending requests (after restarting service) to endpoints mentioned, only getting auth token. Turns out endpoint metrics are dynamically created and do get exposed as expected once added.

So all that is needed is the tenant id tag addition, as planned.

**Comment by tatu-at-datastax** [comment]
Fixed by #1872.
